connect/mirror/src/main/java/org/apache/kafka/connect/mirror/MirrorSourceConnector.java

```diff
@@ -387,4 +402,50 @@ boolean isHeartbeatTopic(String topic) {
     String formatRemoteTopic(String topic) {
         return replicationPolicy.formatRemoteTopic(sourceAndTarget.source(), topic);
     }
+
+    // Writing the translated offsets to the __consumer_offsets topic in the destination cluster,
+    // as long as no consumers in that group are connected to the destination cluster
+    private void syncCommitOffset() throws InterruptedException, ExecutionException {
+        MirrorClient mirrorClient = new MirrorClient(config.targetConsumerConfig());
+        Map<String, Map<TopicPartition, OffsetAndMetadata>> allOffsets =
+            mirrorClient.remoteConsumerOffsets(config.sourceClusterAlias(), config.adminTimeout());
```

Comment:
Can we use the OffsetSyncStore class to do offset translation instead of reading back the checkpoint? The MirrorCheckpointTask class is generating the checkpoint -- why not sync the offsets at the same time?

Reply:
thanks, it indeed saves efforts, will attempt the consumer offset sync in `MirrorCheckpointTask` in the next iteration